test/test_datapipe.py

```diff
@@ -562,6 +562,15 @@ def _worker_init_fn(worker_id):
 
 class TestFunctionalIterDataPipe(TestCase):
 
+    def _serialization_test_helper(self, datapipe, has_two_children=False):
+        serialized_dp = pickle.dumps(datapipe)
+        deserialized_dp = pickle.loads(serialized_dp)
```

Comment:
nit: `deserialized_dp = pickle.loads(pickle.dumps(datapipe))`

Reply:
Would it be good to keep them separate so that it will be more obvious whether `dumps` or `loads` throws an error when one of them fails?

Reply:
But, you have already tested it before invoking them. See comment below.